### PR TITLE
New Kedge structure = new documentation

### PR DESCRIPTION
**Pull request by cdrage**
This reorganizes and rewrites many parts of the file reference to
reflect the new format of Kedge (separate keys for each controller).

Tables are now correctly formatted as well as some minor reoganizational
/ structural changes to the documentation.

### CODE REVIEW

**Comment by cdrage** [comment]
Ping @kadel (this is the last TODO from your PR #565 )

File: docs/file-reference.md [hunk 1]
```diff
@@ -89,62 +93,68 @@ name: <string>
...
+| name              | string                                                     | yes      | The name of the app or microservice this particular file defines.                                                                                                  |
+| appversion        | string                                                     | no       | The version of the app or microservice this particular file defines.                                                                                               |
+| labels            | object                                                     | no       | Map of string keys and values that can be used to organize and categorize (scope and select) objects. May match selectors of replication controllers and services. |
+| deployments       | array of [deploymentObject](#deploymentObject)             | no       | [deploymentObject](#deploymentObject)                                                                                                                              |
```
[the remark this answers is not in the document]
Comment (cdrage) [reply docs/file-reference.md]: Yeah... I always forget that :-1: updated!

**Comment by cdrage** [comment]
@kadel  I seem to always forget that links don't work unless it's lowercase in markdown. Changes pushed.

**Comment by kadel** [comment]
:+1: nice work @cdrage

**Pull request merged**